sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -2136,98 +2138,128 @@ case class GpuStringInstr(str: Expression, substr: Expression)
+    fromCv: GpuColumnVector,
+    toCv: GpuColumnVector): ColumnVector = {
+    if (ansiEnabled &&
+      NumberConverter.isConvertOverflowCvCvCv(strCv.getBase, fromCv.getBase, toCv.getBase)) {
+      ConvUtils.overflowInConvError()
     }
+    NumberConverter.convertCvCvCv(strCv.getBase, fromCv.getBase, toCv.getBase)
   }
 
-  override def convertToGpu(
-    numStr: Expression,
-    fromBase: Expression,
-    toBase: Expression): GpuExpression = GpuConv(numStr, fromBase, toBase)
-}
-
-
-case class GpuConv(num: Expression, fromBase: Expression, toBase: Expression)
-  extends GpuTernaryExpression {
-
   override def doColumnar(
-    v1: GpuColumnVector,
-    v2: GpuColumnVector,
-    v3: GpuColumnVector): ColumnVector = {
-    throw new UnsupportedOperationException()
+      strS: GpuScalar, fromCv: GpuColumnVector, toCv: GpuColumnVector): ColumnVector = {
+    if (!strS.isValid) {
+      GpuColumnVector.columnVectorFromNull(fromCv.getRowCount.toInt, dataType)
+    } else {
+      if (ansiEnabled &&
+        NumberConverter.isConvertOverflowSCvCv(strS.getBase, fromCv.getBase, toCv.getBase)) {
+        ConvUtils.overflowInConvError()
+      }
+      NumberConverter.convertSCvCv(strS.getBase, fromCv.getBase, toCv.getBase)
+    }
   }
 
-  override def doColumnar(v1: GpuScalar, v2: GpuColumnVector, v3: GpuColumnVector): ColumnVector = {
-    throw new UnsupportedOperationException()
-  }
+  override def doColumnar(
+      strS: GpuScalar, fromS: GpuScalar, toCv: GpuColumnVector): ColumnVector = {
 
-  override def doColumnar(v1: GpuScalar, v2: GpuScalar, v3: GpuColumnVector): ColumnVector = {
-    throw new UnsupportedOperationException()
+    if (!strS.isValid || !fromS.isValid) {
+      GpuColumnVector.columnVectorFromNull(toCv.getRowCount.toInt, dataType)
+    } else {
+      fromS.getValue match {
+        case fromRadix: Int =>
+          if (ansiEnabled &&
+          NumberConverter.isConvertOverflowSSCv (strS.getBase, fromRadix, toCv.getBase) ) {
+            ConvUtils.overflowInConvError()
+          }
+          NumberConverter.convertSSCv(strS.getBase, fromRadix, toCv.getBase)
+        case _ => throw new UnsupportedOperationException()
+      }
+    }
   }
 
-  override def doColumnar(v1: GpuScalar, v2: GpuColumnVector, v3: GpuScalar): ColumnVector = {
-    throw new UnsupportedOperationException()
+  override def doColumnar(
+      strS: GpuScalar, fromCv: GpuColumnVector, toS: GpuScalar): ColumnVector = {
+    if (!strS.isValid || !toS.isValid) {
+      GpuColumnVector.columnVectorFromNull(fromCv.getRowCount.toInt, dataType)
+    } else {
+      toS.getValue match {
+        case toRadix: Int =>
+          if (ansiEnabled &&
+            NumberConverter.isConvertOverflowSCvS(strS.getBase, fromCv.getBase, toRadix)) {
+            ConvUtils.overflowInConvError()
+          }
+          NumberConverter.convertSCvS(strS.getBase, fromCv.getBase, toRadix)
+        case _ => throw new UnsupportedOperationException()
+      }
```

Comment:
```suggestion
      val toRadix = toS.getValue.asInstance[Int]
      if (ansiEnabled &&
          NumberConverter.isConvertOverflowSCvS(strS.getBase, fromCv.getBase, toRadix)) {
        ConvUtils.overflowInConvError()
      }
      NumberConverter.convertSCvS(strS.getBase, fromCv.getBase, toRadix)
```

Reply:
Report compile error: asInstance is not a member of Any
The type of toS.getValue is Any.

Reply:
sorry for the typo, it should be asInstanceOf